stan/math/prim/fun/typedefs.hpp

```diff
@@ -28,6 +28,32 @@ using vector_d = Eigen::Matrix<double, Eigen::Dynamic, 1>;
  */
 using row_vector_d = Eigen::Matrix<double, 1, Eigen::Dynamic>;
 
+/**
+ * Types used for profiling.
+ */
+struct profile_meta {
```

Comment:
I think just make it one big struct, and turn it into a class so you can have default constructors.